samcli/commands/sync/command.py

```diff
@@ -365,17 +377,17 @@ def do_cli(
                     on_failure=None,
                 ) as deploy_context:
                     with SyncContext(
-                        dependency_layer, build_context.build_dir, build_context.cache_dir
+                        dependency_layer, build_context.build_dir, build_context.cache_dir, skip_deploy_sync
                     ) as sync_context:
                         if watch:
                             execute_watch(
-                                template_file,
-                                build_context,
-                                package_context,
-                                deploy_context,
-                                sync_context,
-                                dependency_layer,
-                                code,
+                                template=template_file,
+                                build_context=build_context,
+                                package_context=package_context,
+                                deploy_context=deploy_context,
+                                sync_context=sync_context,
+                                auto_dependency_layer=dependency_layer,
+                                disable_infra_syncs=code,
```

Comment:
Parameter is wrong? Shouldn't this be `skip_deploy_sync`?

Reply:
This is when both code and watch are provided to sync. This will completely disable the infra sync right?

Reply:
Sorry, my bad! I mismatched with the new option that we are adding since they are very similar

Reply:
np! Have a note added already to avoid confusion in the future. https://github.com/aws/aws-sam-cli/blob/143e0b654c1b83062f4f318979ab059f81646b87/samcli/commands/sync/command.py#L524-L526